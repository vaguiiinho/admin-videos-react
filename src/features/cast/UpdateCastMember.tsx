import { Box, Paper, Typography } from "@mui/material";
import { useSnackbar } from "notistack";
import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { CastMember } from "../../types/CastMembers";
import { initialState, useGetCastMemberQuery, useUpdateCastMemberMutation } from "./castMemberSlice";
import { CastMemberForm } from "./components/CastMemberForm";

export function UpdateCastMember() {

    const id = useParams().id || ""
    const { data: castMember, isFetching } = useGetCastMemberQuery({ id });


    const [updateCastMember, status] = useUpdateCastMemberMutation()

    const [castMemberState, setCastMemberState] = useState<CastMember>(initialState)

    const { enqueueSnackbar } = useSnackbar()

    async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
        e.preventDefault();
        await updateCastMember(castMemberState);
    }


    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setCastMemberState({ ...castMemberState, [name]: name === "type" ? Number(value) : value, });
    };

    useEffect(() => {
        if (castMember) {
            setCastMemberState(castMember.data);
        }
    }, [castMember]);



    useEffect(() => {
        if (status.isSuccess) {
            enqueueSnackbar('Cast Member updated!', { variant: "success" })
        }
        if (status.error) {
            enqueueSnackbar('Error updating cast member!', { variant: "error" })
        }

    }, [enqueueSnackbar, status.error, status.isSuccess])

    return (
        <Box>
            <Paper>
                <Box p={2}>
                    <Box mb={2}>
                        <Typography variant="h4">
                            Update Category
                        </Typography>
                    </Box>
                </Box>
                <CastMemberForm
                    handleSubmit={handleSubmit}
                    handleChange={handleChange}
                    isLoading={status.isLoading}
                    castMember={castMemberState}
                    isDisabled={status.isLoading}
                />
            </Paper>
        </Box>
    )
}
