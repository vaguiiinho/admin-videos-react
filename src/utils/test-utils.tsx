import { configureStore } from '@reduxjs/toolkit'
import type { RenderOptions } from '@testing-library/react'
import { render } from '@testing-library/react'
import React, { PropsWithChildren } from 'react'
import { Provider } from 'react-redux'

import type { RootState } from '../app/store'
// As a basic setup, import your same slice reducers
import { apiSlice } from '../features/api/apiSlice'
import { SnackbarProvider } from 'notistack'
import { BrowserRouter } from 'react-router-dom'

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: Partial<RootState>
    store?: ReturnType<typeof configureStore>
}

export function renderWithProviders(
    ui: React.ReactElement,
    extendedRenderOptions: ExtendedRenderOptions = {}
) {
    const {

        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                [apiSlice.reducerPath]: apiSlice.reducer,
            },

        }),
        ...renderOptions
    } = extendedRenderOptions

    const Wrapper = ({ children }: PropsWithChildren) => (
        <Provider store={store}>
            <BrowserRouter>
                <SnackbarProvider>
                    {children}
                </SnackbarProvider>
            </BrowserRouter>
        </Provider>
    )

    // Return an object with the store and all of RTL's query functions
    return {
        store,
        ...render(ui, { wrapper: Wrapper, ...renderOptions })
    }
}