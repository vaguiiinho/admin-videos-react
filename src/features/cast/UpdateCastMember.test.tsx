
import { renderWithProviders } from "../../utils/test-utils";
import { UpdateCastMember } from "./UpdateCastMember";


describe("UpdateCastMember", () => {
    it("should render correctly", () => {
        const { asFragment } = renderWithProviders(<UpdateCastMember />);
        expect(asFragment()).toMatchSnapshot();
    });
});