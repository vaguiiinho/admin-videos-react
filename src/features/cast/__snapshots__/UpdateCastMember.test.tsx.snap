// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateCastMember should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
          >
            Update Category
          </h4>
        </div>
      </div>
      <div
        class="MuiBox-root css-19midj6"
      >
        <form>
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-ipzchf-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r0:"
                    id=":r0:-label"
                  >
                    Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r0:"
                      name="name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-81qg8w"
                      >
                        <span>
                          Name *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
              >
                <label
                  class="MuiFormLabel-root MuiFormLabel-colorPrimary css-nwsdb5-MuiFormLabel-root"
                >
                  Type
                </label>
                <div
                  aria-labelledby="type of cast member"
                  class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
                  role="radiogroup"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input css-j8yymo"
                        name="type"
                        type="radio"
                        value="1"
                      />
                      <span
                        class="css-1qiat4j"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                          data-testid="RadioButtonUncheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z8nmqa-MuiSvgIcon-root"
                          data-testid="RadioButtonCheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                    >
                      Director
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-j8yymo"
                        name="type"
                        type="radio"
                        value="2"
                      />
                      <span
                        class="css-1qiat4j"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                          data-testid="RadioButtonUncheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                          data-testid="RadioButtonCheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                    >
                      Actor
                    </span>
                  </label>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
            >
              <div
                class="MuiBox-root css-10egq61"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                  href="/cast-members"
                  tabindex="0"
                >
                  Back
                </a>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-dk52k6-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Save
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
